control/injector/Injector.php

```diff
@@ -585,15 +585,56 @@ public function inject($object, $asType=null) {
 		$objtype = $asType ? $asType : get_class($object);
 		$mapping = isset($this->injectMap[$objtype]) ? $this->injectMap[$objtype] : null;
 
+		$spec = empty($this->specs[$objtype]) ? array() : $this->specs[$objtype];
+
 		// first off, set any properties defined in the service specification for this
 		// object type
-		if (isset($this->specs[$objtype]) && isset($this->specs[$objtype]['properties'])) {
+		if(!empty($spec['properties']) && is_array($spec['properties'])) {
 			foreach ($this->specs[$objtype]['properties'] as $key => $value) {
 				$val = $this->convertServiceProperty($value);
 				$this->setObjectProperty($object, $key, $val);
 			}
 		}
 
+		// Populate named methods
+		if (!empty($spec['calls']) && is_array($spec['calls'])) {
+			foreach ($spec['calls'] as $method) {
+				// Ignore any blank entries from the array; these may be left in due to config system limitations
+				if(!$method) continue;
+
+				// Format validation
+				if(!is_array($method) || !isset($method[0]) || isset($method[2])) {
```

Comment:
I think this won't quite work for length 1 arrays. Because it will always error due to the last isset. But besides that, make sure to change the index from 2 to 1